### ISSUE TITLE
httputil.ReverseProxy logs to Go default logger instead of logrus

### ISSUE THREAD

**Issue by cyrilc-pro** [traefik/traefik#10543]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

Traefik should always log using logrus.

### What did you see instead?

On certain circumstances, Traefik, through the use of `httputil.ReverseProxy`, logs messages using Go default logger. For instance:
```
2024/03/06 08:00:38 httputil: ReverseProxy read error during body copy: stream error: stream ID 1255; CANCEL; received from peer
```

### What version of Traefik are you using?

v2.11

### What is your environment & configuration?

[not relevant]

**Comment by cyrilc-pro** [comment]
If accepted, I can propose PR's for fixing this in 2.x and 3.x.
The basic idea would be to use something like: `log.SetOutput(logger.WriterLevel(logrus.InfoLevel))`

**Comment by rtribotte** [comment]
Hello @cyrilc-pro,

Thanks for reporting this!

> If accepted, I can propose PR's for fixing this in 2.x and 3.x.
The basic idea would be to set field ErrorLog when creating the httputil.ReverseProxy instance:

Sure, we would gladly welcome a PR against v2.11 branch for this, thanks!

**Comment by peacewalker122** [comment]
Hi, I'd love to work on this issue

@rtribotte @cyrilc-pro

**Comment by rtribotte** [comment]
Hi @peacewalker122, sure! I have assigned you.

**Comment by kevinpollet** [comment]
Closed by #11344
